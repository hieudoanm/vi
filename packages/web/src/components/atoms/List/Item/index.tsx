import React from 'react';

export type ItemProps = {
  last?: boolean;
};

const Item: React.FC<ItemProps> = ({ last, children }) => (
  <div className={last ? '' : 'border-b'}>{children}</div>
);

Item.displayName = 'List.Item';
Item.defaultProps = { last: false };

export default Item;
