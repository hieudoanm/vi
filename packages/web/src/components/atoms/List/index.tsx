import React from 'react';
import Item from './Item';

export type ListProps = {
  className?: string;
};

const List: React.FC<ListProps> = ({ className = '', children }) => {
  return (
    <div className={`overflow-hidden rounded-lg border ${className}`}>
      {children}
    </div>
  );
};

List.displayName = 'List';
List.defaultProps = { className: '' };

export default Object.assign(List, { Item });
