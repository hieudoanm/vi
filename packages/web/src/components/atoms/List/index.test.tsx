import List from '.';
import { render } from '@testing-library/react';

describe('List', () => {
  test('render default', () => {
    const wrapper = render(<List>List</List>);
    expect(wrapper.container).toMatchSnapshot();
  });
});
