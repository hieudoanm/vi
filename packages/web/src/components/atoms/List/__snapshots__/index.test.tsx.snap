// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List render default 1`] = `
<div>
  <div
    class="overflow-hidden rounded-lg border "
  >
    List
  </div>
</div>
`;
